improver/calibration/ensemble_calibration.py

```diff
@@ -1263,6 +1287,44 @@ class ApplyEMOS(PostProcessingPlugin):
     Class to calibrate an input forecast given EMOS coefficients
     """
 
+    @staticmethod
+    def _get_attribute(coefficients, attribute_name):
+        """Get the value for the requested attribute, ensuring that the
+        attribute is present consistently across the cubes within the
+        coefficients cubelist.
+
+        Args:
+            coefficients (iris.cube.CubeList):
+                EMOS coefficients
+            attribute_name (str):
+                Name of expected attribute
+
+        Returns:
+            None or Any:
+                Returns None if the attribute is not present. Otherwise,
+                the value of the attribute is returned.
+
+        Raises:
+            ValueError: If coefficients do not share the expected attributes.
+        """
+        attributes = [
+            tuple(c.attributes[attribute_name])
+            for c in coefficients
+            if c.attributes.get(attribute_name) is not None
+        ]
+
+        if not attributes:
+            return None
```

Comment:
Is there a check anywhere in case the attribute is just undefined on all input cubes, e.g. no distribution attribute in any of them? Currently it will just return None which will then be set as the distribution.

Reply:
Good point. I've added some handling if the attribute requested is completely missing from all the coefficients cubes.